### ISSUE TITLE
False positive 'unnecessary parenthesis' with 'if let':

### ISSUE THREAD

**Issue by Aaron1011** [rust-lang/rust#75300]
The following code:

```rust
fn main() {
    if let true = (true && false) {}
}
```

produces the following warning:

```
warning: unnecessary parentheses around `let` scrutinee expression
 --> src/main.rs:2:19
  |
2 |     if let true = (true && false) {}
  |                   ^^^^^^^^^^^^^^^ help: remove these parentheses
  |
  = note: `#[warn(unused_parens)]` on by default
```

However, removing the parenthesis results in an error:

```
   Compiling playground v0.0.1 (/playground)
error[E0658]: `let` expressions in this position are experimental
 --> src/main.rs:2:8
  |
2 |     if let true = true && false {}
  |        ^^^^^^^^^^^^^^^
  |
  = note: see issue #53667 <https://github.com/rust-lang/rust/issues/53667> for more information
  = help: add `#![feature(let_chains)]` to the crate attributes to enable

error: `let` expressions are not supported here
 --> src/main.rs:2:8
  |
2 |     if let true = true && false {}
  |        ^^^^^^^^^^^^^^^
  |
  = note: only supported directly in conditions of `if`- and `while`-expressions
  = note: as well as when nested within `&&` and parenthesis in those conditions
```

**Comment by Aaron1011** [comment]
Closing as duplicate of https://github.com/rust-lang/rust/issues/60336
